engine/src/execution_graph/logic_controllers/PhysicalPlanGenerator.h

```diff
@@ -314,8 +300,28 @@ struct tree_processor {
 			} else {
 				auto child_kernel_type = child->kernel_unit->get_type_id();
 				auto parent_kernel_type = parent->kernel_unit->get_type_id();
-				if ((child_kernel_type == kernel_type::JoinPartitionKernel && parent_kernel_type == kernel_type::PartwiseJoinKernel)
-					    || (child_kernel_type == kernel_type::SortAndSampleKernel &&	parent_kernel_type == kernel_type::PartitionKernel)
+				if (child_kernel_type == kernel_type::JoinPartitionKernel && parent_kernel_type == kernel_type::PartwiseJoinKernel) {
+
+					std::size_t max_data_load_concat_cache_bytes_size = 400000000; // 400 MB
```

Comment:
shouldnt this be configurable?

Reply:
It is, its JOIN_PARTITION_SIZE_THRESHOLD